### ISSUE TITLE
Switch Language

### ISSUE THREAD

**Issue by nenad007** [Philipp15b/php-i18n#63]
Is there a solution, how I can switch language during script execution? 

I have the case where I send Emails/Notifications to the users by a cron script, depending on the defined user Language every User should got the Email/Notification in the correct language.

**Comment by Philipp15b** [comment]
No, that's not possible. However, I'm open to accepting pull requests that add this feature while keeping API stability.

**Comment by nenad007** [comment]
I have implemented this functionality, but I did not found a way to keep the API in full, concretely I lost calls like L::HELLO which must be called by one of: L::{lang}_HELLO, L::HELLO(), L('HELLO') ... 

and instead to have one cached file per Language I generate one file for all languages where I set {lang} as prefix to the constants. I'm not sure if this is a problem as the file got cached in Memory anyway.

If you have some idea how I could do it better please let me know and I will go this way and make a pull request.

**Comment by TheJosh** [comment]
The language compiler could compile each language class with a separate class name (which includes the language code) and then the main `L` class could use magic methods to delegate into the currently selected language class... I might do some experimentation with this.

**Comment by TheJosh** [comment]
Yep it's not possible, without either (a) changing the api or (b) using runkit.

**Comment by yaberkane05** [comment]
Would it be possible to switch language by dropping the instance of i18n object, force language, then re init() ?

**Comment by tangramor** [comment]
I have similar requirement, so I created my own i18n lib based on this one: https://github.com/tangramor/php-i18n
